### DISCUSSION TITLE
How to query against the data model instead of the response model?

### DISCUSSION THREAD

**Discussion by drmcclelland** [AutoMapper/AutoMapper.Extensions.OData#178] [Q&A]
My question is regarding the `GetAsync` method:

```
public static async Task<ICollection<TModel>> GetAsync<TModel, TData>(this IQueryable<TData> query, IMapper mapper, ODataQueryOptions<TModel> options, HandleNullPropagationOption handleNullPropagation = HandleNullPropagationOption.Default);
```

Would there be a way to pass in `ODataQueryOptions<TData> options` instead?

The reason is for us to be able to perform a query that filters based on a property that is part of the **data model** (TData) but is NOT part of the **response model** (TModel).

**Comment by BlaiseD** [comment]
[EF.Property](https://learn.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.ef.property?view=efcore-7.0) might help if you're using EF.  Otherwise you may want to post an example showing how you would do this outside of AutoMapper.
